src/middleware.js

```diff
@@ -56,26 +56,31 @@ export default function wrapper(context) {
         return;
       }
 
-      if (!res.get('Content-Type')) {
+      if (!res.getHeader('Content-Type')) {
         // content-type name(like application/javascript; charset=utf-8) or false
         const contentType = mime.contentType(path.extname(filename));
-
-        if (contentType) {
-          res.set('Content-Type', contentType);
-        }
+        res.setHeader(
+          'Content-Type',
+          contentType || 'application/octet-stream'
+        );
```

Comment:
We should **always** use official middleware API, so for example we shoudl use `express` API, for connect (no example API) we should use http module API, why? Because other middlewares/tools can add additional logic for built-in API, and we will not working properly with them

Reply:
So we need to create `compatibility` file and put some utils there, like:
```

Reply:
I agree with your explanation, and that's why this middleware can't be used by Koa, even though they expose native request/response.

However, if we scope our support level to vanilla HTTP, express, connect, and other frameworks that don't add extra logic, then we don't need to do that. We can't possibly support all frameworks anyway.

Reply:
So let's take my suggestion, it is easy and will not confusing for future refactoring

Reply:
If the support level is strictly express and connect, I think it's unnecessary.

Going back to the case with [`compression`](https://www.npmjs.com/package/compression), I'm not convinced why a third-party middleware requires this abstraction when a first-party middleware doesn't.

Reply:
Can you clarify?